portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +34,74 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages;
+
+  public void init() {
+      //submitTime = new Date();
+      messages = new ArrayList<String>();
+  }
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+  throws IOException {
+
+      Query query = new Query("Task//");//.addSort("timestamp", SortDirection.DESCENDING);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+    
+      for (Entity entity : results.asIterable()) {
+          long id = entity.getKey().getId();
+          String message = (String) entity.getProperty("message");
+          messages.add(message);
+      }
+
+      response.setContentType("application/json;");
+      response.getWriter().println(convertToJsonUsingGson(messages));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String clientMessage = request.getParameter("clientMessage");
+      long timeStamp = System.currentTimeMillis();
+
+      /*System.out.println(htmlMessage);
+      messages.add(htmlMessage);
+      Gson gson = new Gson();
+      String json = convertToJson(clientMessage));*/
+      Entity messageEntity = new Entity("Messages");
+      messageEntity.setProperty("message", clientMessage);
+      messageEntity.setProperty("timestamp", timeStamp);
+
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(messageEntity);
+
+      response.sendRedirect("/index.html");
+
+    }
+
+  private String convertToJsonUsingGson(ArrayList<String> messageList) {
+      Gson gson = new Gson();
+      String json = gson.toJson(messageList);
+    
+      return json;
+  }
+
+  private String convertToJson(ArrayList<String> messageList) {
+      String json = "[ { ";
+      for(int i = 0; i<(messageList.size());i= i+2){
+          json += " \"message\" : ";
+          json += "\""+messageList.get(i)+"\", ";
+          json += " \"timestamp\" : ";
+          json += "\""+messageList.get(i+1)+"\"";
```

Comment:
This implementation assumes size of message is even number, is this true?

Reply:
I included this method for when I want to start sorting comments using other fields such as name or timestamp, so when a comment is submitted it will either contain 2 fields(timestamp and message) or 3 field (timestamp, message, and name) that it will be sorted by. I might go in and change some things within this method, or just delete it completely depending on what the next steps calls us to do.

Reply:
So if it contains 3 fields, won't you get error on line 89?

Reply:
I deleted the entire method.